ml3d/datasets/shapenet.py

```diff
@@ -0,0 +1,207 @@
+        self.task = task
+
+        self.cat = {}
+        self.catfile = os.path.join(self.dataset_path,
+                                    'synsetoffset2category.txt')
+        with open(self.catfile, 'r') as f:
+            for idx, line in enumerate(f):
+                ls = line.strip().split()
+                self.cat[idx] = ls[1]
+
+        self.meta = {}
+        for item in self.cat:
+            self.meta[item] = []
+            dir_point = os.path.join(self.dataset_path, self.cat[item],
+                                     'points')
+            dir_seg = os.path.join(self.dataset_path, self.cat[item],
+                                   'points_label')
+            fns = sorted(os.listdir(dir_point))
+            for fn in fns:
+                token = (os.path.splitext(os.path.basename(fn))[0])
+                self.meta[item].append(
+                    (join(dir_point,
+                          token + '.pts'), join(dir_seg, token + '.seg')))
+
+        splits = []
+        splits_path = join(self.dataset_path, 'train_test_split')
+        for split in [
+                'shuffled_train_file_list.json', 'shuffled_test_file_list.json',
+                'shuffled_val_file_list.json'
+        ]:
+            with open(join(splits_path, split)) as source:
+                json_source = source.read()
+                splits.append(
+                    [i.split('/')[-1] for i in json.loads(json_source)])
+        train_split, test_split, val_split = splits
+
+        self.all_files = []
+        self.train_files = []
+        self.val_files = []
+        self.test_files = []
+        for item in self.cat:
+            for fn in self.meta[item]:
+                self.all_files.append((item, fn[0], fn[1]))
+                file = fn[0].split('/')[-1].split('.')[0]
+                if file in train_split:
+                    self.train_files.append((item, fn[0], fn[1]))
+                elif file in test_split:
+                    self.test_files.append((item, fn[0], fn[1]))
+                else:
+                    self.val_files.append((item, fn[0], fn[1]))
+        shuffle(self.train_files)
+        shuffle(self.test_files)
+        shuffle(self.val_files)
+
+    @staticmethod
+    def get_label_to_names(task="classification"):
+        if task == "classification":
+            label_to_names = {
+                0: 'Airplane',
+                1: 'Bag',
+                2: 'Cap',
+                3: 'Car',
+                4: 'Chair',
+                5: 'Earphone',
+                6: 'Guitar',
+                7: 'Knife',
+                8: 'Lamp',
+                9: 'Laptop',
+                10: 'Motorbike',
+                11: 'Mug',
+                12: 'Pistol',
+                13: 'Rocket',
+                14: 'Skateboard',
+                15: 'Table'
+            }
+        elif task == "segmentation":
+            raise NotImplementedError
```

Comment:
I see that reading segmentation labels is also supported. But this will throw an error for `segmentation` task.
Can you add this part too?

Reply:
I agree that this is not ideal, though it won't throw an error when loading the dataset regardless of the task, as it is never actually passed to the function, so the default value is used.

I wasn't able to find a reference to the actual object part segmentation labels, as they are different for each object category. From the [official website](https://shapenet.cs.stanford.edu/iccv17/): "Each category is annotated with 2 to 6 parts and there are 50 different parts annotated in total."

Reply:
I've added dummy labels.

Reply:
That'll work, thanks.